File: 이코테/P92_큰수의법칙/P92.py
N, M, K = map(int, input().split())

big1 = 0
big2 = 0

for i in list(map(int, input().split())):
    
    if big1 < i:
        big2 = big1
        big1 = i
        
    elif big2 < i:
        big2 = i

times = M//(K+1)
remain = M%(K+1)

print(big1 * remain + (big1 * K + big2)* times)

File: 이코테/P92_큰수의법칙/P92.c
#include<stdio.h>

int main(){

    int N, M, K;
    int big1 = 0, big2 = 0;

    scanf("%d %d %d",&N,&M,&K);

    for(int i = 0 ; i < N ; i++)
    {
        int temp;
        scanf("%d", &temp);

        if(temp > big1)
        {
            big2 = big1;
            big1 = temp;
        }
        else if(temp >big2)
        {
            big2 = temp;
        }
    }

    int times = 8/(K+1);
    int remain = 8%(K+1);

    printf("%d\n", remain * big1 + times * (big1 * K + big2));

}